File: calc/app/src/main/java/com/firz/calc/MainActivity.java
package com.firz.calc;

import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import java.text.BreakIterator;

public class MainActivity extends AppCompatActivity {

    TextView textHasil;
    EditText bil1, bil2;
    double hasil = 0;

    public void load() {
        textHasil = findViewById(R.id.textHasil);
        bil1 = findViewById(R.id.bil1);
        bil2 = findViewById(R.id.bil2);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });

        load();
    }

    public void jumlah(View view) {
        if (bil1.getText().toString().equals("") || bil2.getText().toString().equals("")) {
            Toast.makeText(this, "Ada Kolom Kosong", Toast.LENGTH_SHORT).show();
        } else {
            double var1 = Double.parseDouble(bil1.getText().toString());
            double var2 = Double.parseDouble(bil2.getText().toString());
            hasil = var1 + var2;

            textHasil.setText(hasil + "");
        }
    }
    public void kurang(View view) {
        if (bil1.getText().toString().equals("") || bil2.getText().toString().equals("")) {
            Toast.makeText(this, "Ada Kolom Kosong", Toast.LENGTH_SHORT).show();
        } else {
            double var1 = Double.parseDouble(bil1.getText().toString());
            double var2 = Double.parseDouble(bil2.getText().toString());
            hasil = var1 - var2;

            textHasil.setText(hasil + "");
        }
    }
    public void kali(View view) {
        if (bil1.getText().toString().equals("") || bil2.getText().toString().equals("")) {
            Toast.makeText(this, "Ada Kolom Kosong", Toast.LENGTH_SHORT).show();
        } else {
            double var1 = Double.parseDouble(bil1.getText().toString());
            double var2 = Double.parseDouble(bil2.getText().toString());
            hasil = var1 * var2;

            textHasil.setText(hasil + "");
        }
    }
    public void bagi(View view) {
        if (bil1.getText().toString().equals("") || bil2.getText().toString().equals("")) {
            Toast.makeText(this, "Ada Kolom Kosong", Toast.LENGTH_SHORT).show();
        } else {
            double var1 = Double.parseDouble(bil1.getText().toString());
            double var2 = Double.parseDouble(bil2.getText().toString());
            hasil = var1 / var2;

            textHasil.setText(hasil + "");
        }
    }
}

File: LaravelAPI/app/src/main/java/com/firz/laravelapi/activity/MainActivity.kt
package com.firz.laravelapi.activity

import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.Menu
import android.view.View
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.SearchView
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import androidx.recyclerview.widget.DefaultItemAnimator
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.swiperefreshlayout.widget.SwipeRefreshLayout
import com.firz.laravelapi.R
import com.firz.laravelapi.activity.tambahData.TambahMahasiswaActivity
import com.firz.laravelapi.adapter.MahasiswaAdapter
import com.firz.laravelapi.config.NetworkConfig
import com.firz.laravelapi.databinding.ActivityMainBinding
import com.firz.laravelapi.model.DataMahasiswa
import com.firz.laravelapi.model.ResponseListMahasiswa
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class MainActivity : AppCompatActivity(), SwipeRefreshLayout.OnRefreshListener {
    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        enableEdgeToEdge()
        ViewCompat.setOnApplyWindowInsetsListener(binding.root) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

        val btnTambah = binding.btnTambah
        btnTambah.setOnClickListener {
            val tambah = Intent(this, TambahMahasiswaActivity::class.java)
            startActivity(tambah)
        }

        val swipeRefresh = binding.swipeRefreshLayout
        swipeRefresh.setOnRefreshListener(this)

        val appbar = findViewById<androidx.appcompat.widget.Toolbar>(R.id.app_bar_search)
        setSupportActionBar(appbar)
        getMahasiswa()
    }

    private fun getMahasiswa() {
        NetworkConfig().getService().getMahasiswa()
            .enqueue(object : Callback<ResponseListMahasiswa> {
                override fun onResponse(
                    call: Call<ResponseListMahasiswa>,
                    response: Response<ResponseListMahasiswa>
                ) {
                    this@MainActivity.binding.progressIndicator.visibility = View.GONE
                    if (response.isSuccessful) {
                        val receiveDatas = response.body()?.data
                        setToAdapter(receiveDatas)
                    }
                    binding.swipeRefreshLayout.isRefreshing = false
                }

                override fun onFailure(call: Call<ResponseListMahasiswa>, t: Throwable) {
                    this@MainActivity.binding.progressIndicator.visibility = View.GONE
                    Log.d("Retrofit onFailure: ", "onFailure: ${t.stackTrace}")

                    binding.swipeRefreshLayout.isRefreshing = false
                }

            })
    }

    private fun cariMahasiswa(query: String?) {
        this.binding.progressIndicator.visibility = View.GONE
        NetworkConfig().getService().cariMahasiswa(query)
            .enqueue(object : Callback<ResponseListMahasiswa> {
                override fun onResponse(
                    call: Call<ResponseListMahasiswa>,
                    response: Response<ResponseListMahasiswa>
                ) {
                    this@MainActivity.binding.progressIndicator.visibility = View.GONE
                    if (response.isSuccessful) {
                        val receiveDatas = response.body()?.data
                        setToAdapter(receiveDatas)
                    }
                }

                override fun onFailure(call: Call<ResponseListMahasiswa>, t: Throwable) {
                    this@MainActivity.binding.progressIndicator.visibility = View.GONE
                    Log.d("Retrofit onFailure: ", "onFailure: ${t.stackTrace}")
                }

            })
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        menuInflater.inflate(R.menu.menu, menu)
        val cariItem = menu?.findItem(R.id.app_bar_search)
        val cariView: SearchView = cariItem?.actionView as SearchView
        cariView.setOnQueryTextListener(object : SearchView.OnQueryTextListener{
            override fun onQueryTextSubmit(query: String?): Boolean {
                cariMahasiswa(query)
                return true
            }

            override fun onQueryTextChange(newText: String?): Boolean {
                return true
            }

        })
        return true
    }

    private fun setToAdapter(receiveDatas: List<DataMahasiswa?>?) {
        val adapter = MahasiswaAdapter(receiveDatas)
        val lm = LinearLayoutManager(this)
        this.binding.rvMahasiswa.layoutManager = lm
        this.binding.rvMahasiswa.itemAnimator = DefaultItemAnimator()
        this.binding.rvMahasiswa.adapter = adapter
    }

    override fun onRefresh() {
        getMahasiswa()
    }
}
